__tests__/commands/install/focus.js

```diff
@@ -224,6 +224,30 @@ test.concurrent('focus works correctly when focusing on a scoped package', (): P
   });
 });
 
+test.concurrent('focus work even when the workspace root also depends on other workspaces', (): Promise<void> => {
+  return runInstall(
+    {focus: true},
+    {source: 'focus-root-dep', cwd: '/packages/example-yarn-workspace-1'},
+    async (config, reporter) => {
+      const existAndIsMaterialized = async moduleDir => {
+        expect(await fs.exists(path.join(moduleDir, 'package.json'))).toBeTruthy();
+        const stat = await fs.lstat(moduleDir);
+        expect(stat.isSymbolicLink()).toEqual(false);
+      };
+
+      // The one in the root must exist, and must be materialized (since we're focusing on example-yarn-workspace-1)
+      await existAndIsMaterialized(path.join(config.cwd, '..', '..', 'node_modules', 'example-yarn-workspace-2'));
```

Comment:
this check will always fail because it's expecting it to not be a symlink even though it is one.

Reply:
Why should it be a symlink? If you focus on a workspace I thought it meant that all other workspaces would be fetched from the network.

Reply:
The implementation is actually a little different from that. When you do a focused install, it installs everything like normal, and only adds in remote installs "shallowly". That means it doesn't put the remote copy of `example-yarn-workspace-2` at root/node_modules. That is still a symlink to the local copy. Instead it installs it at `packages/example-yarn-workspace-1/node_modules/examples-yarn-workspace-2`. That helps deal with version conflict issues and also makes it quicker to switch back and forth between focused and not focused.

Reply:
so basically the shallow install comes first in node module resolution so it masks the symlink and causes it not to be used.